micall/tests/test_kive_watcher.py

```diff
@@ -3160,3 +3160,42 @@ def test_collate_csv_with_sample_already_filled():
     KiveWatcher.extract_csv(source2, target, 'ignored', source_count=1)
 
     assert target.getvalue() == expected_target
+
+def test_launch_main_good_pipeline_id(mock_open_kive, default_config):
+    mock_session = mock_open_kive.return_value
+    kive_watcher = KiveWatcher(default_config)
+    kive_watcher.app_urls = {
+        default_config.micall_filter_quality_pipeline_id: '/containerapps/102'}
+    kive_watcher.app_args = {
+        default_config.micall_filter_quality_pipeline_id: dict(
+            quality_csv='/containerargs/103')}
+
+    inputs = {'quality_csv': {'url': '/datasets/104', 'id': 104}}
+    run_batch = {'url': '/batches/101'}
+    kive_watcher.find_or_launch_run(pipeline_id=42,
+                                    inputs=inputs,
+                                    run_name='MiCall filter quality on 140101_M01234',
+                                    run_batch=run_batch)
+
+def test_launch_main_bad_pipeline_id(mock_open_kive, default_config):
+    mock_session = mock_open_kive.return_value
+    kive_watcher = KiveWatcher(default_config)
+    kive_watcher.app_urls = {
+        default_config.micall_filter_quality_pipeline_id: '/containerapps/102'}
+    kive_watcher.app_args = {
+        default_config.micall_filter_quality_pipeline_id: dict(
+            quality_csv='/containerargs/103')}
+
+    inputs = {'quality_csv': {'bad_argument': 777, 'id': 104}}
+    run_batch = {'url': '/batches/101'}
+    pipeline_id = 42
+
+    with pytest.raises(ValueError) as excinfo:
+        kive_watcher.find_or_launch_run(pipeline_id=pipeline_id,
+                                        inputs=inputs,
+                                        run_name='MiCall filter quality on 140101_M01234',
+                                        run_batch=run_batch)
+
+    assert f'The specified app with id {pipeline_id}' \
```

Comment:
It's slightly easier to use the `match` parameter in [`pytest.raises`]. Notice that it's a regular expression, so using the `r` prefix lets you use backslashes for character classes or when you really want to match a period.

[`pytest.raises`]: https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions

Reply:
Done